### PR TITLE
Chore: generic better error_handling for API calls

### PR DESCRIPTION
**Pull request by k9ert**
For page rendering, we have a lot of generic error_handling code in the controller. This frees us from handling errors individually in controller endpoints.

The same is missing for API-Endpoints. In order to do that, we need:
1. Add the header `Accept: application/json` to each fetch call
2. Implement  Error-handling code in the error_handlers which check for that header and `return {"error": some_error}`
3. check for that field in the json-response on the client side and `showError`
4. Remove individual error_handling code in API endpoints

This PR is starting that process. (2.) is already done in this PR. Here is a list of individual endpoints where this is implemented:
* toggle_hide_sensitive_info
* togglePriceCheck
* fetchAddressData

... and tested:

- [ ] toggle_hide_sensitive_info
- [ ] togglePriceCheck
- [ ] fetchAddressData

This is a Draft and it's not yet clear how to test these changes consistently.

### CODE REVIEW

File: src/cryptoadvance/specter/templates/components/price_bar.jinja [hunk 1]
```diff
@@ -203,12 +203,21 @@
             }
             let url = `{{ url_for('price_endpoint.toggle') }}`;
             var formData = new FormData(document.getElementById('price-form'));
+            result = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
+            if (result) {
+                location.reload();
+                return
+            };
+            
             try {
```
Comment (k9ert) [inline src/cryptoadvance/specter/templates/components/price_bar.jinja:212]: Delete from here

File: src/cryptoadvance/specter/templates/includes/address-data.html [hunk 2]
```diff
@@ -84,110 +84,91 @@ <h1>{{ _("Address details") }}</h1><br>
...
-                        <tr><td>{{ _("Label") }}:</td><td><address-label data-copy-hidden="true" data-address="${address}" data-wallet="${this.wallet}" data-label="${this.label}" data-service-id="${this.serviceId}"></address-label><br></td></tr>
-                        <tr><td>{{ _("From wallet") }}:</td><td><a href=${walletLink}>${walletName}<a></td></tr>
-                        <tr><td>{{ _("Address index") }}:</td><td>${addressIndex}</td></tr>
-                        <tr><td>{{ _("Is change address") }}:</td><td>${isChange ? '{{ _("Yes") }}' : '{{ _("No") }}'}</td></tr>
-                        <tr><td>{{ _("Used") }}:</td><td>${this.used ? 'Yes' : 'No'}</td></tr>
-                        <tr><td>{{ _("UTXO count") }}:</td><td>${this.utxo}</td></tr>
-                        <tr><td>{{ _("Amount") }}:</td><td>${this.amount} <span class="amount-price note ${this.amountPrice ? '' : 'hidden'}">${this.amountPrice}</span></td></tr>
-                    `;
+            jsonResponse = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/includes/address-data.html:87]:
```suggestion
            let jsonResponse = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```

File: src/cryptoadvance/specter/templates/base.jinja [hunk 3]
```diff
@@ -124,29 +124,12 @@
...
-						location.reload();
-						return
-					};
-					showError(`{{ _("Failed to toggle sensitive info mode...") }}`)
-				} catch(e) {
-					console.log("Caught error: ", e);
-					showError(e);
-				}
+				
+				result = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/base.jinja:128]: Seems to be general thing: It doesn't seem that the result / response vars have been declared before

Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/base.jinja:128]:
```suggestion
				let result = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```

File: src/cryptoadvance/specter/templates/components/price_bar.jinja [hunk 4]
```diff
@@ -203,27 +203,10 @@
...
-                const jsonResponse = await response.json();
-                if (jsonResponse.success) {
-                    location.reload();
-                    return
-                };
-                showError(`{{ _("Failed to toggle price mode...") }}`)
-            } catch(e) {
-                console.log("Caught error: ", e);
-                showError(e);
+            result = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/components/price_bar.jinja:206]:
```suggestion
            let result = await send_request(url, 'POST', "{{ csrf_token() }}", formData);
```

File: src/cryptoadvance/specter/templates/includes/address-data.html (hunk 2 above)
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/includes/address-data.html:87]: Also, are those quotation marks correct or should it be back ticks:
`` `{{ csrf_token() }}` ``
Similar to sth. like this:

let url = `` `{{ url_for('auth_endpoint.toggle_hide_sensitive_info') }}` ``

File: src/cryptoadvance/specter/templates/components/price_bar.jinja (hunk 4 above)
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/components/price_bar.jinja]: Fixed in https://github.com/cryptoadvance/specter-desktop/pull/2014/commits/f354b0b10e1e4c817dce3e044f9e3c5a815f0dc3

File: src/cryptoadvance/specter/templates/base.jinja (hunk 3 above)
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/base.jinja]: Fixed in https://github.com/cryptoadvance/specter-desktop/pull/2014/commits/f354b0b10e1e4c817dce3e044f9e3c5a815f0dc3

Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/base.jinja]: Fixed in https://github.com/cryptoadvance/specter-desktop/pull/2014/commits/f354b0b10e1e4c817dce3e044f9e3c5a815f0dc3

File: src/cryptoadvance/specter/templates/includes/address-data.html (hunk 2 above)
Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/includes/address-data.html]: Fixed in https://github.com/cryptoadvance/specter-desktop/pull/2014/commits/f354b0b10e1e4c817dce3e044f9e3c5a815f0dc3

**Review by moneymanolis — APPROVED** [review]

**Pull request merged**